cmd/promtool/tsdb.go

```diff
@@ -398,21 +398,24 @@ func openBlock(path, blockID string) (*tsdb.DBReadOnly, tsdb.BlockReader, error)
 	if err != nil {
 		return nil, nil, err
 	}
-	blocks, err := db.Blocks()
-	if err != nil {
-		return nil, nil, err
-	}
+
 	var block tsdb.BlockReader
+
+	if blockID == "" {
+		blockID, err = db.LastBlockID(nil)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
 	if blockID != "" {
-		for _, b := range blocks {
-			if b.Meta().ULID.String() == blockID {
-				block = b
-				break
-			}
+		b, err := db.Block(nil, filepath.Join(blockID))
```

Comment:
Why do we need the `filepath.Join`?

Reply:
After reviewing the code, I realized that filepath.Join was not necessary in this particular case, so I removed it and made the necessary adjustments to the code.